### ISSUE TITLE
Issue with Reproduction -- Awaiting Response

### ISSUE THREAD

**Issue by ShuoYang-1998** [JosephKJ/OWOD#18]
Hi authors,
I failed to re-produce the reported ORE performance. I couldn't even reproduce the results of faster-rcnn + fine-tuning baseline. In task 3, I got 32 and 11 for previous and current ap50. Compared to the results in your paper, the t3 prev and current is 37 and 12 respectively. The t3 log.txt [t3_val_test_log.txt](https://github.com/JosephKJ/OWOD/files/6179642/t3_val_test_log.txt) is attached, could you kindly help to fix the problem?
Thanks!

**Comment by JosephKJ** [comment]
Hi @ShuoYang-1998, thanks for sharing the log. It is interesting to see that you have much better A-OSE than what we report in the paper. Have you tried the same number of iterations that we had shared?

Further, did you use the pre-trained models that we shared, or have you trained it from scratch?

**Comment by ShuoYang-1998** [comment]
I trained exactly the same iterations as your provided model_backup. I tried both training from scratch and using your provided pre-trained models, but I cannot get the same results as in your paper in both cases. The attached files [t4_log.txt](https://github.com/JosephKJ/OWOD/files/6181755/t4_log.txt) is the t4_log when evaluating your provided models, you can see that I got Known AP50: 24.88 while the result in your paper is 26.66. Additionally, in this [t2_log.txt](https://github.com/JosephKJ/OWOD/files/6181768/t2_log.txt) , you can see that I got AOSE of 8714 while in your paper is 7772. Note the above two log files were both generated by using your provided pre-trained models. When I try to re-train, the performance gap becomes more unacceptable. Am I doing something wrong? 


,

**Comment by JosephKJ** [comment]
What machine are you using, from the logs I presume it is also a DGX-2.

**Comment by ShuoYang-1998** [comment]
8 * V100 16g

**Comment by JosephKJ** [comment]
Let me check whether the models shared were correct.

**Comment by ShuoYang-1998** [comment]
Hi, I uploaded the experimental script [run_OWOD.txt](https://github.com/JosephKJ/OWOD/files/6201644/run_OWOD.txt) and all config files [OWOD_configs.zip](https://github.com/JosephKJ/OWOD/files/6201647/OWOD_configs.zip)  along with the running log [log.txt](https://github.com/JosephKJ/OWOD/files/6201650/log.txt). As shown in the log, the re-training results are far away from the reported results. It would be much appreciated if you can help me fix the problem. Thanks!

**Comment by AmingWu** [comment]
@ShuoYang-1998 @JosephKJ ，When I run the code, I obtained the result. Is it similar to your results?

python tools/train_net.py --num-gpus 4 --config-file configs/OWOD/t1/t1_train.yaml SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.005

![QQ截图20210328135054](https://user-images.githubusercontent.com/38445175/112743791-ec91a380-8fcc-11eb-85ce-291152772754.png)

**Comment by ShuoYang-1998** [comment]
> @ShuoYang-1998 @JosephKJ ，When I run the code, I obtained the result. Is it similar to your results?
> 
> python tools/train_net.py --num-gpus 4 --config-file configs/OWOD/t1/t1_train.yaml SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.005
> 
> ![QQ截图20210328135054](https://user-images.githubusercontent.com/38445175/112743791-ec91a380-8fcc-11eb-85ce-291152772754.png)
> 
> 

@AmingWu You should run train-val-test, and report the test results.

**Comment by AmingWu** [comment]
@ShuoYang-1998 
![QQ截图20210328142823](https://user-images.githubusercontent.com/38445175/112744388-45b00600-8fd2-11eb-93cf-da63a88de0d9.png)
，Thanks for your reply. This is the test result. Is it similar to you?

**Comment by ShuoYang-1998** [comment]
> @ShuoYang-1998 
> ![QQ截图20210328142823](https://user-images.githubusercontent.com/38445175/112744388-45b00600-8fd2-11eb-93cf-da63a88de0d9.png)
> ，Thanks for your reply. This is the test result. Is it similar to you?
> 
> 

The task1 is not class incremental, can you share your task3 results?

**Comment by AmingWu** [comment]
OK，Now I run the task3, Does this order is correct?

I first run:

python tools/train_net.py --num-gpus 4 --config-file configs/OWOD/t3/t3_train.yaml SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.005

Then I run:

python tools/train_net.py --num-gpus 4 --eval-only --config-file configs/OWOD/t1/t1_test.yaml SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.005

**Comment by ShuoYang-1998** [comment]
> OK，Now I run the task3, Does this order is correct?
> 
> I first run:
> 
> python tools/train_net.py --num-gpus 4 --config-file configs/OWOD/t3/t3_train.yaml SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.005
> 
> Then I run:
> 
> python tools/train_net.py --num-gpus 4 --eval-only --config-file configs/OWOD/t1/t1_test.yaml SOLVER.IMS_PER_BATCH 4 SOLVER.BASE_LR 0.005

I think it should be train-ft-val-test, the task1 doesn't need ft. You can refer to my uploaded 'run_OWOD.txt', which contains all training and testing scripts. But I don't know if the script is correct because the author didn't provide running scripts yet.

**Comment by ShuoYang-1998** [comment]
@JosephKJ Some people also has the reproduce problem. Can you fix it？Or you will keep ignoring it?

**Comment by salman-h-khan** [comment]
Hi Shuo, 

The first author has been traveling recently and will soon provide the rerun models. 

Best,
Salman

**Comment by JosephKJ** [comment]
Continuing the discussion in #26

**Comment by iFighting** [comment]
Hi @JosephKJ 
I also can not reproduce your result reported in your paper. I try to run your code several times, but failed to get the same results. 
other people also meet the issue，i really doubt the authenticity  of this work, so i think you should consider the issue seriously.

**Comment by JosephKJ** [comment]
I have added '[replicate.py](https://github.com/JosephKJ/OWOD/blob/master/replicate.sh)' to replicate results from the pertained models shared before. You can find the binaries and logs [here](https://drive.google.com/drive/folders/1Sr4_q0_m2f2SefoebB25Ix3N1VIAua0w), if you want to verify the authenticity of the results. 

Please find my results below:
<img width="660" alt="Replicated Results" src="https://user-images.githubusercontent.com/4231550/117539609-1953c480-b029-11eb-94d5-f77006f15a5c.png">

**Comment by iFighting** [comment]
> I have added '[replicate.py](https://github.com/JosephKJ/OWOD/blob/master/replicate.sh)' to replicate results from the pertained models shared before. You can find the binaries and logs [here](https://drive.google.com/drive/folders/1Sr4_q0_m2f2SefoebB25Ix3N1VIAua0w), if you want to verify the authenticity of the results.
> 
> Please find my results below:
> <img alt="Replicated Results" width="660" src="https://user-images.githubusercontent.com/4231550/117539609-1953c480-b029-11eb-94d5-f77006f15a5c.png">

The result from  the pertained models in your figure is not consistent with the result in your paper。
In addition，we still cannot reproduce the result from the training scheduler。
I think you should serious consider this problem。

**Comment by dyabel** [comment]
> @ShuoYang-1998
> ![QQ截图20210328142823](https://user-images.githubusercontent.com/38445175/112744388-45b00600-8fd2-11eb-93cf-da63a88de0d9.png)
> ，Thanks for your reply. This is the test result. Is it similar to you?

I have the similar result to you in task 1, the mAP is only 52

**Comment by JosephKJ** [comment]
@dyabel : How many number of GPUs are you using?

**Comment by dyabel** [comment]
> @dyabel : How many number of GPUs are you using?

I used 4 gpus and I have found the reason, I should multiple the iters and step by 2. But the WI in task1 is still 0.05092. I have attached my config.yaml and log.txt.
[dyabel.zip](https://github.com/JosephKJ/OWOD/files/6476004/dyabel.zip)

 and

**Comment by JosephKJ** [comment]
Are the other metrics matching up?

**Comment by dyabel** [comment]
> Are the other metrics matching up?

yes

**Comment by dyabel** [comment]
> > Are the other metrics matching up?
> 
> yes

I mean i can only reproduce task 1 except WI. Task 2-4 do not match.

**Comment by akshitac8** [comment]
Hello @JosephKJ @yuandhu I was successfully able to reproduce the results and also attached the table for the same

![Screenshot from 2021-09-13 11-41-53](https://user-images.githubusercontent.com/21353757/133043671-99f5b371-3452-4269-aedb-5d44dcc82214.png)
